### PR TITLE
 fix: remove incorrect uncaughtPromiseRejection error

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

TL;DR:- Remove `uncaughtPromiseRejection:` from the error where the issue is of a different type. 

For example, when the user makes a type error in the async field like `button.onClick`. Currently, the user notices a `UncaughtPromiseRejection` error instead of TypeError or ReferenceError.

This PR makes sure to show relevant TypeError or ReferenceError for such cases. 

## Type of change

- Bug fix

## How Has This Been Tested?

- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by ohansFavour — APPROVED** [review]

**Review by ohansFavour — CHANGES_REQUESTED** [review]
Let's be clear on the requirements here, as this only renames the error messages.

**Review by ohansFavour — DISMISSED** [review]

**Comment by ramsaptami** [comment]
LGTM, tested using framework functions where a promise is returned and error codes now indicate the correct area of error instead of UncaughtPromiseRejection.
CC: @Rishabh-Rathod

**Review by ohansFavour — APPROVED** [review]
